### DISCUSSION TITLE
Provide generic version of JsonSerializable attribute

### DISCUSSION THREAD

**Discussion by HakamFostok** [dotnet/runtime#91852] [Ideas]
Right now the `JsonSerializable` is taking a type parameter like this

```
[JsonSerializable(typeof(string))]
```

What about adding a new attribute like this
```
[JsonSerializable<string>()]
```

**Comment by gfoidl** [comment]
The suggested way is to create an API-request according to [API Review Process](https://github.com/dotnet/runtime/blob/main/docs/project/api-review-process.md), then this request can be handled.

**Reply by HakamFostok** [reply]
thank you so much for the advice, here is the issue
please feel free to comment or edit on that issue if you have any note, or I miss anything required
https://github.com/dotnet/runtime/issues/91869
